src/js/facility-locator/containers/VAMap.jsx

```diff
@@ -1,34 +1,151 @@
+    const { currentQuery } = this.props;
+    const newQuery = nextProps.currentQuery;
+
+    if (currentQuery.position !== newQuery.position) {
+      this.updateUrlParams({
+        location: `${nextProps.currentQuery.position.latitude},${nextProps.currentQuery.position.longitude}`,
+      });
+    }
+  }
+
+  // pushes coordinates to URL so that map link is useful for sharing
+  // TODO (bshyong): try out existing query-string npm library
+  updateUrlParams(params) {
+    const { location } = this.props;
+
+    const queryParams = map({
+      ...location.query,
+      ...params,
+    }, (v, k) => {
+      return `${k}=${v}`;
+    }).join('&');
+    browserHistory.push(`${location.pathname}?${queryParams}`);
+  }
+
+  // takes obj of form {latitude: 0, longitude: 0}
+  reverseGeocode(position) {
+    mapboxClient.geocodeReverse(position, {
+      types: 'address',
+    }, (err, res) => {
+      // TODO (bshyong): handle error case
+      const placeName = res.features[0].place_name;
+      this.props.updateSearchQuery({
+        searchString: placeName,
+      });
+      this.updateUrlParams({
+        address: placeName,
+      });
+    });
+  }
+
+  handleSearch = () => {
+    const { currentQuery } = this.props;
+
+    this.updateUrlParams({
+      address: currentQuery.searchString,
+      location: `${currentQuery.position.latitude},${currentQuery.position.longitude}`,
+    });
+    this.props.search(currentQuery);
   }
 
   render() {
-    const position = [38.8976763, -77.03653];
+  // defaults to White House coordinates initially
+    const coords = this.props.currentQuery.position;
+    const position = [coords.latitude, coords.longitude];
+
     return (
-      <Map ref="map" center={position} zoom={13} >
-        <TileLayer
-            url="http://{s}.tile.osm.org/{z}/{x}/{y}.png"
-            attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'/>
-        <Marker position={position}>
-          <Popup>
-            <span>A pretty CSS3 popup.<br/>Easily customizable.</span>
-          </Popup>
-        </Marker>
-      </Map>
+      <div>
+        <div className="columns medium-3">
+          <ResultsPane onSearch={this.handleSearch}/>
+        </div>
+        <div className="medium-9 columns">
+          <Map ref="map" center={position} zoom={13} >
+            <TileLayer
+                url={`https://api.mapbox.com/styles/v1/mapbox/streets-v9/tiles/256/{z}/{x}/{y}?access_token=${mapboxToken}`}
+                attribution='Map data &copy; <a href="http://openstreetmap.org">OpenStreetMap</a> contributors, <a href="http://creativecommons.org/licenses/by-sa/2.0/">CC-BY-SA</a>, Imagery © <a href="http://mapbox.com">Mapbox</a>'/>
+            <Marker position={position}>
+              <Popup>
+                <span>You are here.</span>
+              </Popup>
+            </Marker>
+            <TownHall position={[38.8983294, -77.0295762]}>
+              <span>Town hall</span>
```

Comment:
Definitely. This component was just a test—I'll remove it from this display in the interim.